src/diffusers/models/modeling_utils.py

```diff
@@ -315,7 +316,12 @@ def save_pretrained(
         weights_name = _add_variant(weights_name, variant)
         weight_name_split = weights_name.split(".")
         if len(weight_name_split) in [2, 3]:
-            weights_name_pattern = weight_name_split[0] + "{suffix}." + ".".join(weight_name_split[1:])
+            if variant is not None:
+                weights_name_pattern = (
+                    weight_name_split[0] + f".{variant}" + "{suffix}." + ".".join(weight_name_split[2:])
+                )
+            else:
+                weights_name_pattern = weight_name_split[0] + "{suffix}." + ".".join(weight_name_split[1:])
```

Comment:
To satisfy the diffusion model prefix format in `huggingface_hub`.

Reply:
Sure. 

`split_torch_state_dict_into_shards()` [here](https://github.com/huggingface/diffusers/blob/48e36353d8cbf0322ec1ad0684b95d11f70af2de/src/diffusers/models/modeling_utils.py#L344C28-L344C62) needs a `filename_pattern`. 

When we're dealing with `variant`, we need to put it such that the pattern becomes:

```
'diffusion_pytorch_model.fp16{suffix}.safetensors'
```

This `suffix` will ultimately be automatically replaced by `huggingface_hub` appropriately depending on whether the given state dict is sharded or not.

Reply:
ok can we do similar to what's in transformers https://github.com/huggingface/transformers/blob/5af7d41e49bbfc8319f462eb45253dcb3863dfb7/src/transformers/modeling_utils.py#L2741